ratis-common/src/main/java/org/apache/ratis/protocol/PauseLeaderElectionRequest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.protocol;
+
+public final class PauseLeaderElectionRequest extends RaftClientRequest{
+
+  private final RaftPeerId server;
+  private final boolean pause;
+
+  public PauseLeaderElectionRequest(
+      ClientId clientId, RaftPeerId serverId, RaftGroupId groupId, long callId, long timeoutMs, boolean pause) {
+    super(clientId, serverId, groupId, callId, readRequestType(), timeoutMs);
+    this.pause = pause;
+    this.server = serverId;
+  }
+
+  public RaftPeerId getPausedServer() {
```

Comment:
We already have getServerId() in RaftClientRequest so that this method is not needed.